src/thor/costmatrix.cc

```diff
@@ -1243,9 +1243,10 @@ std::string CostMatrix::RecostFormPath(GraphReader& graphreader,
   auto source_vertex = PointLL{source_edge.ll().lng(), source_edge.ll().lat()};
   auto target_vertex = PointLL{target_edge.ll().lng(), target_edge.ll().lat()};
   std::vector<PointLL> points;
-  for (const auto& path_edge : path_edges) {
-    auto is_first_edge = path_edge == path_edges.front();
-    auto is_last_edge = path_edge == path_edges.back();
```

Comment:
for the case of seemingly trivial routes which go against the oneway, both of these are true and then further down it'd trim the shape according to a trivial route which is not doing the right thing to do. we have to do this index based to really tell if it's the first/last edge.

Reply:
yeah these needed to be pointer comparison not equality operator